Cuda 11 build fails

Config:
```bash
cmake -DKokkos_ARCH_POWER9=ON -DKokkos_ARCH_VOLTA70=True -DKokkos_ENABLE_CUDA=True -DKokkos_ENABLE_OPENMP=True -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER=${PARTHENON_ROOT}/external/Kokkos/bin/nvcc_wrapper -DPARTHENON_DISABLE_MPI=On ${PARTHENON_ROOT}
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 6.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /ccs/home/pgrete/src/parthenon/external/Kokkos/bin/nvcc_wrapper
-- Check for working CXX compiler: /ccs/home/pgrete/src/parthenon/external/Kokkos/bin/nvcc_wrapper -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at cmake/Format.cmake:36 (message):
  Couldn't determine clang-format version.  clang-format 8.0 is required -
  results on other versions may not be stable
Call Stack (most recent call first):
  CMakeLists.txt:50 (include)


-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5") found components:  CXX 
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Found HDF5: /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/hdf5-1.10.4-qj7qoumocmjj2xkwbwotazo22sugeq5o/lib/libhdf5.so;/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/zlib-1.2.11-fvgnqf6k3ffhltldndu7pmntzvoyfsk4/lib/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so (found version "1.10.4") found components:  C 
-- Setting default Kokkos CXX standard to 11
-- Setting policy CMP0074 to use <Package>_ROOT variables
-- The project name is: Kokkos
-- Using -std=c++11 for C++11 standard as feature
-- Execution Spaces:
--     Device Parallel: CUDA
--     Host Parallel: OPENMP
--       Host Serial: NONE
-- 
-- Architectures:
--  POWER9
--  VOLTA70
-- Found TPLLIBDL: /usr/lib64/libdl.so  
-- Gold standard is up-to-date. No download required.
-- Building unit tests.
-- Building integration tests.
-- Building performance tests.
-- Building regression tests.
-- MPI not found, not building regression tests with mpi
-- MPI not found, not building coverage regression tests with mpi
-- MPI not found, not building regression tests with mpi
-- MPI not found, not building coverage regression tests with mpi
-- MPI not found, not building regression tests with mpi
-- MPI not found, not building coverage regression tests with mpi
-- MPI not found, not building regression tests with mpi
-- MPI not found, not building coverage regression tests with mpi
-- PAR_LOOP_LAYOUT='MANUAL1D_LOOP' (default par_for wrapper layout)
-- PAR_LOOP_INNER_LAYOUT='TVR_INNER_LOOP' (default par_for_inner wrapper layout)
-- Configuring done
-- Generating done
-- Build files have been written to: /gpfs/alpine/ast146/world-shared/build-cuda-latest
```

Build:
```
[ 67%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o
/ccs/home/pgrete/src/parthenon/src/interface/container.hpp(293): internal error: assertion failed at: "overload.c", line 22098 in find_top_temporary


1 catastrophic error detected in the compilation of "/ccs/home/pgrete/src/parthenon/src/bvals/cc/bvals_cc.cpp".
Compilation aborted.
nvcc error   : 'cudafe++' died due to signal 6 
nvcc error   : 'cudafe++' core dumped
make[2]: *** [src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o] Error 134
make[1]: *** [src/CMakeFiles/parthenon.dir/all] Error 2
make: *** [all] Error 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@
 - [[PR 303]](https://github.com/lanl/parthenon/pull/303) Changed `Mesh::BlockList` from a `std::list<MeshBlock>` to a `std::vector<std::shared_ptr<MeshBlock>>`, making `FindMeshBlock` run in constant, rather than linear, time. Loops over `block_list` in application drivers must be cahnged accordingly.
 
 ### Fixed (not changing behavior/API/variables/...)
+- [[PR 310]](https://github.com/lanl/parthenon/pull/310) Fix Cuda 11 builds.
 
 ### Removed (removing behavior/API/varaibles/...)
 

--- a/src/interface/container.hpp
+++ b/src/interface/container.hpp
@@ -142,7 +142,7 @@ class Container {
   // Queries related to CellVariable objects
   //
   const CellVariableVector<T> &GetCellVariableVector() const { return varVector_; }
-  const MapToCellVars<T> GetCellVariableMap() const { return varMap_; }
+  const MapToCellVars<T> &GetCellVariableMap() const { return varMap_; }
   CellVariable<T> &Get(std::string label) {
     auto it = varMap_.find(label);
     if (it == varMap_.end()) {
@@ -164,8 +164,8 @@ class Container {
   //
   // Queries related to SparseVariable objects
   //
-  const SparseVector<T> GetSparseVector() const { return sparseVector_; }
-  const MapToSparse<T> GetSparseMap() const { return sparseMap_; }
+  const SparseVector<T> &GetSparseVector() const { return sparseVector_; }
+  const MapToSparse<T> &GetSparseMap() const { return sparseMap_; }
   SparseVariable<T> &GetSparseVariable(const std::string &label) {
     auto it = sparseMap_.find(label);
     if (it == sparseMap_.end()) {
@@ -194,7 +194,7 @@ class Container {
   // Queries related to FaceVariable objects
   //
   const FaceVector<T> &GetFaceVector() const { return faceVector_; }
-  const MapToFace<T> GetFaceMap() const { return faceMap_; }
+  const MapToFace<T> &GetFaceMap() const { return faceMap_; }
   FaceVariable<T> &GetFace(std::string label) {
     auto it = faceMap_.find(label);
     if (it == faceMap_.end()) {

